src/commands/PublishCommand.js

```diff
@@ -292,7 +292,7 @@ export default class PublishCommand extends Command {
       const version = this.updatesVersions[depName];
 
       if (deps[depName] && version) {
-        deps[depName] = "^" + version;
+        deps[depName] = this.flags.exact ? version : "^" + version;
```

Comment:
I think to understand properly you need to know arity of operators, so it can become confusing for some people. You can read it both as 
```
(this.flags.exact ? version : "^") + version;
```
and
```
this.flags.exact ? version : ("^" + version);
```